tests/snippets/set.py

```diff
@@ -99,17 +99,64 @@ def __hash__(self):
 assert a == set([1,2,3,4,5])
 assert_raises(TypeError, lambda: a.update(1))
 
+a = set([1,2,3])
+b = set()
+for e in a:
+	assert e == 1 or e == 2 or e == 3
+	b.add(e)
+assert a == b
+
+a = set([1,2,3])
+a |= set([3,4,5])
+assert a == set([1,2,3,4,5])
+try:
+	a |= 1
+except TypeError:
+	pass
+else:
+	assert False, "TypeError not raised"
+
 a = set([1,2,3])
 a.intersection_update([2,3,4,5])
 assert a == set([2,3])
 assert_raises(TypeError, lambda: a.intersection_update(1))
 
+a = set([1,2,3])
+a &= set([2,3,4,5])
+assert a == set([2,3])
+try:
+	a &= 1
+except TypeError:
+	pass
+else:
+	assert False, "TypeError not raised"
```

Comment:
Could you use the `assert_raises` function for these checks? Just makes these a bit less verbose

Reply:
You cant do assignment inside lambda function so the best I can think about is something like this:
```py
a = [set([1,2,3])]
def f():
    a[0] |= set([4,5])
assert_raises(TypeError, f)
```
I believe the current option is more clear. Any other suggestion?

Reply:
Oh, I completely missed that it's an assignment! Hopefully in the future we'll have something like `assertRaises` from `unittest`:

```python
with self.assertRaises(TypeError):
    a[0] |= set([4,5])
```

Reply:
The with statement should work, so we could do the same now by implementing a contextmanager.